WordPress/gutenberg
# Site Editor: Don't display welcome guide in hybrid theme

Fixes #55857

## What?

This PR prevents classic themes with `block-template-parts` support from displaying the welcome guide when opening the template parts editor.

![template-parts](https://github.com/WordPress/gutenberg/assets/54422211/9be145d1-7270-4c94-97d8-bf474b2fc55b)

## Why?

Everything explained in the welcome guide is related to block themes. With hybrid themes, you cannot edit the entire site, and you cannot use the global style UI.

## How?
If it is not a block theme, the welcome guide will not be displayed. In the Site Editor, there are also [welcome guides related to Templates, Global Styles and Pages](https://github.com/WordPress/gutenberg/blob/d137227a67c43d04a6cba0cfacf8dd7f95cb74d5/packages/edit-site/src/components/welcome-guide/index.js), but if the current theme is not a block theme, you should not be able to access these pages in the first place.

## Testing Instructions

Note: If you have previously hidden the Welcome Guide, please follow the steps below to reset your test environment (`localhost:8889`) settings.

Delete the preference from database: 

```
npm run wp-env run tests-cli wp db query "DELETE FROM wp_usermeta WHERE meta_key = 'wp_persisted_preferences'"
```

Delete `WP_PREFERENCES_USER_X` from the browser's local storage:

![local-strage](https://github.com/WordPress/gutenberg/assets/54422211/80b6e07b-a596-41e3-bc7e-f04e328af861)

---

- Open `http://localhost:8889/wp-admin/`.
- Activate Emptyhybryd theme
- Access to Appearance > Template Parts > header > Editor canvas.
- The welcome guide should NOT be displayed.
- Activate Twenty Twenty-Four.
- Access to Appearance > Editor > Editor canvas.
- The welcome guide should be displayed. 

### e2e test

- Revert only the `packages/edit-site/src/components/welcome-guide/editor.js` file to the trunk version.
- Run `npm run test:e2e:playwright test/e2e/specs/site-editor/hybrid-theme.spec.js`
- The test should fail.
